src/adaptations/device-layer/include/Weave/DeviceLayer/WeaveDeviceConfig.h

```diff
@@ -673,4 +673,109 @@
 #define WEAVE_DEVICE_CONFIG_DEFAULT_TUNNEL_TELEMETRY_INTERVAL_MS 300000
 #endif
 
+// -------------------- Event Logging Configuration --------------------
+
+/**
+ * @def WEAVE_DEVICE_CONFIG_EVENT_LOGGING_CRIT_BUFFER_SIZE
+ *
+ * @brief
+ *   A size, in bytes, of the individual critical event logging buffer.
+ *   Note: the critical event buffer must exist.
+ */
+#ifndef WEAVE_DEVICE_CONFIG_EVENT_LOGGING_CRIT_BUFFER_SIZE
+#define WEAVE_DEVICE_CONFIG_EVENT_LOGGING_CRIT_BUFFER_SIZE (1024)
+#endif
+
+#if (WEAVE_DEVICE_CONFIG_EVENT_LOGGING_CRIT_BUFFER_SIZE <= 0)
+#error "The Prod critical event buffer must exist (WEAVE_DEVICE_CONFIG_EVENT_LOGGING_CRIT_BUFFER_SIZE > 0)"
+#endif
+
+/**
+ * @def WEAVE_DEVICE_CONFIG_EVENT_LOGGING_PROD_BUFFER_SIZE
+ *
+ * @brief
+ *   A size, in bytes, of the individual production event logging buffer.
+ *   Note: the production event buffer must exist.
+ */
+#ifndef WEAVE_DEVICE_CONFIG_EVENT_LOGGING_PROD_BUFFER_SIZE
+#define WEAVE_DEVICE_CONFIG_EVENT_LOGGING_PROD_BUFFER_SIZE (512)
+#endif
+
+#if (WEAVE_DEVICE_CONFIG_EVENT_LOGGING_PROD_BUFFER_SIZE <= 0)
+#error "The Prod event buffer must exist (WEAVE_DEVICE_CONFIG_EVENT_LOGGING_PROD_BUFFER_SIZE > 0)"
+#endif
+
+/**
+ * @def WEAVE_DEVICE_CONFIG_EVENT_LOGGING_INFO_BUFFER_SIZE
+ *
+ * @brief
+ *   A size, in bytes, of the individual info event logging buffer.
+ *   Note: set to 0 to disable info event buffer.
```

Comment:
Setting this to 0 doesn't just disable the buffer, it disables all support for Info level events, right?

The same comment applies to WEAVE_DEVICE_CONFIG_EVENT_LOGGING_DEBUG_BUFFER_SIZE, and to the descriptions in ESP32 Kconfig.

Reply:
Updated description